src/features/auth/schema.ts

```diff
@@ -1,8 +1,16 @@
 import {z} from "zod";
 
-export const basic = z.object({
+export const login = z.object({
   email: z.string().email(),
   password: z.string().min(8),
 });
 
-export type BasicPayload = z.infer<typeof basic>;
+export const user = z.object({
+  email: z.string().email(),
+  username: z.string(),
+  firstname: z.string(),
+  lastname: z.string(),
+  bio: z.string() 
```

Comment:
bio should be `.optional()`-ed, same for `username`,
also include the ctx where you use it in the variable name, like: `loginUser`, `signupUser`, `updateUser`, ...